### PR TITLE
Support javascript files with type=module (close #1725)

### LINKED ISSUE #1725: Support javascript files with type=module
Roadmap:
- [x] es6-module
- [ ] import/export

https://jakearchibald.com/2017/es-modules-in-browsers/

### PR DESCRIPTION
**Pull request by LavrovArtem**

### CODE REVIEW

File: test/server/data/page/src.html [hunk 1]
```diff
@@ -60,6 +60,11 @@
         <!--<script type="text/javascript">-->
         var someScript;
     </script>
+    <script type="module">
```
Comment (miherlosev) [inline test/server/data/page/src.html]: Support the `import.meta.url` property

```js
<html>
    <head>
		<script type="module">
			console.log(import.meta.url); // url of the html pag on which the module was included
		</script>
	</head>
	<body>
	</body>
</html>
```

File: src/client/sandbox/code-instrumentation/index.ts [hunk 2]
```diff
@@ -82,6 +86,26 @@ export default class CodeInstrumentation extends SandboxBase {
...
+
+        nativeMethods.objectDefineProperty(window, INSTRUCTION.getProxyUrl, {
+            value: (url: any, baseUrl?: string) => {
+                const storedBaseUrl    = urlResolver.getBaseUrl(document);
+                const shouldChangeBase = baseUrl && baseUrl !== storedBaseUrl;
+
+                if (shouldChangeBase)
+                    urlResolver.updateBase(baseUrl, document);
+
+                const proxyUrl = getProxyUrl(String(url), { resourceType: stringifyResourceType({ isScript: true }) });
```
Comment (miherlosev) [inline src/client/sandbox/code-instrumentation/index.ts]: `getProxyUrl(String(url),...`
We can move casting to string inside the `getProxyUrl` function.
```js
export function getProxyUrl (url: string, opts: ProxyUrlOptions) {
    url = String(url);
   
    const params: Array<string> = [opts.sessionId];
```

and then simplify all `getProxyUrl(String(url)` in our code.

File: src/processing/resources/index.ts [hunk 3]
```diff
@@ -9,9 +9,10 @@ import { platform } from 'os';
 
 const IS_WIN: boolean = platform() === 'win32';
 const DISK_RE: RegExp = /^[A-Za-z]:/;
+const PROCESSORS      = [pageProcessor, manifestProcessor, scriptProcessor, stylesheetProcessor];
```
Comment (miherlosev) [inline src/processing/resources/index.ts]: `const RESOURCE_PROCESSORS = ...`

File: src/processing/script/index.ts [hunk 4]
```diff
@@ -76,7 +75,11 @@ function getAst (src: string, isObject: boolean): Program {
     src = isObject ? `(${src})` : src;
 
     try {
-        return parse(src, { allowReturnOutsideFunction: true });
+        return parse(src, {
```
Comment (miherlosev) [inline src/processing/script/index.ts]: Create a const with parsing options on the module level.
```js
const PARSING_OPTIONS = {
    allowReturnOutsideFunction:  true,
    allowImportExportEverywhere: true,
    ecmaVersion:                 11
};

```

File: test/server/data/page/src.html (hunk 1 above)
Comment (LavrovArtem) [reply test/server/data/page/src.html]: https://github.com/DevExpress/testcafe-hammerhead/issues/2118

File: src/processing/script/index.ts [hunk 5]
```diff
@@ -27,8 +26,14 @@ const PROCESSED_SCRIPT_RE: RegExp   = new RegExp([
     reEscape(INSTRUCTION.callMethod),
     reEscape(INSTRUCTION.processScript),
     reEscape(INSTRUCTION.processHtml),
-    reEscape(INSTRUCTION.getPostMessage)
+    reEscape(INSTRUCTION.getPostMessage),
+    reEscape(INSTRUCTION.getProxyUrl)
```
Comment (miherlosev) [inline src/processing/script/index.ts]: Add new line after `const` declaration. Why eslint failed?

File: src/utils/url.ts [hunk 6]
```diff
@@ -125,7 +125,7 @@ export function getURLString (url: string): string {
 }
 
 export function getProxyUrl (url: string, opts: ProxyUrlOptions) {
-    let params: any = [opts.sessionId];
+    const params: Array<string> = [opts.sessionId];
```
Comment (miherlosev) [inline src/utils/url.ts]: Stringify url inside the `getProxyUrl` function and simplify this code line - https://github.com/DevExpress/testcafe-hammerhead/blob/630c488aa2f31d721baf86151c7758f25264baec/src/client/sandbox/node/window.ts#L630

**Pull request merged**